test/unit/ui/map_events.test.js

```diff
@@ -585,3 +585,19 @@ test(`Map#on mousedown does not fire subsequent click event if mouse position ch
     map.remove();
     t.end();
 });
+
+test(`Map#on click fires subsequent click event if there is no corresponding mousedown/mouseup event`, (t) => {
+    const map = createMap(t, { clickTolerance: 4 });
+
+    const click = t.spy();
+    map.on('click', click);
+    const canvas = map.getCanvas();
+
+    const MouseEvent = window(canvas).MouseEvent;
+    const event = new MouseEvent('click', {bubbles: true, clientX: 100, clientY: 100});
```

Comment:
Alternatively we could expose a `simulate.fakeClick` utility, but I didn't think we'd want to encourage anyone to re-use that since it only applies to a corner case.